pyrefly/lib/alt/solve.rs

```diff
@@ -1733,7 +1733,7 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
             (Type::TypedDict(typed_dict), Type::Literal(Lit::Str(field_name))) => {
                 let field_name = Name::new(field_name);
                 if let Some(field) = self.typed_dict_field(typed_dict, &field_name) {
-                    if field.read_only {
+                    if field.read_only.is_some() {
```

Comment:
Would prefer a is_read_only() helper method to get the boolean value, to avoid exposing the implementation unnecessarily.